llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21951,7 +21951,9 @@ bool RISCVTargetLowering::lowerDeinterleaveIntrinsicToLoad(
         Intrinsic::riscv_vlseg6, Intrinsic::riscv_vlseg7,
         Intrinsic::riscv_vlseg8};
 
-    unsigned SEW = ResVTy->getElementType()->getScalarSizeInBits();
+    unsigned SEW = ResVTy->getElementType()->isPointerTy()
+                       ? Subtarget.getXLen()
+                       : ResVTy->getElementType()->getScalarSizeInBits();
```

Comment:
Can we use DataLayout instead of hardcoding XLEN?
```suggestion
unsigned SEW = DI->getDataLayout().getTypeSizeInBits(ResVTy->getElementType());
```

Reply:
Sure~ it's a good idea!